File: Lab2.1/DigitalRow.cs
// DigitalRow.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Lab3
{
    public class DigitalRow : Row
    {
        public DigitalRow() : base() { }

        public DigitalRow(string value) : base(value) { }

        public DigitalRow(DigitalRow other) : base(other) { }

        public int CalculateDigitalSum()
        {
            return value.Where(c => char.IsDigit(c))
                        .Sum(c => c - '0');
        }

        public List<int> GetDigits()
        {
            return value.Where(c => char.IsDigit(c))
                        .Select(c => c - '0')
                        .ToList();
        }

        public override string ToString()
        {
            return $"Digital{base.ToString()}, Digital Sum: {CalculateDigitalSum()}";
        }
    }
}

File: Lab2.1cpp/DigitalRow.cpp
// DigitalRow.cpp
#include "DigitalRow.h"
#include <cctype>

DigitalRow::DigitalRow() : Row() {
}

DigitalRow::DigitalRow(const std::string& value) : Row(value) {
}

DigitalRow::DigitalRow(const DigitalRow& other) : Row(other) {
}

DigitalRow::~DigitalRow() {
}

int DigitalRow::calculateDigitalSum() const {
    int sum = 0;
    for (char c : value) {
        if (std::isdigit(c)) {
            sum += (c - '0');
        }
    }
    return sum;
}

std::vector<int> DigitalRow::getDigits() const {
    std::vector<int> digits;
    for (char c : value) {
        if (std::isdigit(c)) {
            digits.push_back(c - '0');
        }
    }
    return digits;
}

std::string DigitalRow::toString() const {
    return "Digital" + Row::toString() + ", Digital Sum: " + std::to_string(calculateDigitalSum());
}
